### DISCUSSION TITLE
Boot Loop of Outline Docker container

### DISCUSSION THREAD

**Discussion by electric-m** [outline/outline#6879] [Self Hosting]
He ythere,

I followed the Thomas Griffin tutorial to get outline up and running - well almost.
When starting the containers, outline crashes and goe sinto boot loop with the following errors:

`
yarn install v1.22.19
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.1.0"
warning @bull-board/koa > koa-views@7.0.2: This package is deprecated, please use the new fork @ladjs/koa-views. Maintenance is supported by Forward Email at https://forwardemail.net ; follow/watch https://github.com/ladjs/koa-views for updates and release changelog
warning @bull-board/koa > koa-router@10.1.1: **IMPORTANT 10x+ PERFORMANCE UPGRADE**: Please upgrade to v12.0.1+ as we have fixed an issue with debuglog causing 10x slower router benchmark performance, see https://github.com/koajs/router/pull/173
warning @bull-board/koa > koa-views > consolidate@0.16.0: Please upgrade to consolidate v1.0.0+ as it has been modernized with several long-awaited fixes implemented. Maintenance is supported by Forward Email at https://forwardemail.net ; follow/watch https://github.com/ladjs/consolidate for updates and release changelog
warning @types/form-data@2.5.0: This is a stub types definition. form-data provides its own type definitions, so you do not need this installed.
warning @types/sanitize-filename@1.6.3: This is a stub types definition. sanitize-filename provides its own type definitions, so you do not need this installed.
warning autotrack > rollup-plugin-node-resolve@3.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-node-resolve.
warning aws-sdk > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning aws-sdk > url > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^2.6.8"
warning babel-plugin-transform-class-properties > babel-runtime > core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
warning jsdom > domexception@4.0.0: Use your platform's native DOMException instead
warning jsdom > abab@2.0.6: Use your platform's native atob() and btoa() methods instead
warning jsdom > data-urls > abab@2.0.6: Use your platform's native atob() and btoa() methods instead
warning Resolution field "qs@6.9.7" is incompatible with requested version "qs@^6.11.0"
warning koa-router@7.4.0: **IMPORTANT 10x+ PERFORMANCE UPGRADE**: Please upgrade to v12.0.1+ as we have fixed an issue with debuglog causing 10x slower router benchmark performance, see https://github.com/koajs/router/pull/173
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.1.0"
warning socket.io-redis@6.1.1: This package has been renamed to '@socket.io/redis-adapter', please see the migration guide here: https://socket.io/docs/v4/redis-adapter/#migrating-from-socketio-redis
warning umzug > @rushstack/ts-command-line > @rushstack/terminal > @rushstack/node-core-library > z-schema@5.0.6: has issues with node 14
warning vite-plugin-pwa > workbox-build > @surma/rollup-plugin-off-main-thread > magic-string > sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
warning @types/dotenv@8.2.0: This is a stub types definition. dotenv provides its own type definitions, so you do not need this installed.
warning @types/emoji-regex@9.2.0: This is a stub types definition. emoji-regex provides its own type definitions, so you do not need this installed.
warning @types/koa-useragent@2.1.2: This is a stub types definition. koa-useragent provides its own type definitions, so you do not need this installed.
warning @types/mermaid@9.2.0: This is a stub types definition. mermaid provides its own type definitions, so you do not need this installed.
warning babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^2.2.0"
warning i18next-parser > vinyl-fs > glob-stream > unique-stream > through2-filter@3.1.0: this package has been deprecated for breaking semver major
warning jest-environment-jsdom > jsdom > abab@2.0.6: Use your platform's native atob() and btoa() methods instead
warning jest-environment-jsdom > jsdom > domexception@4.0.0: Use your platform's native DOMException instead
warning jest-environment-jsdom > jsdom > data-urls > abab@2.0.6: Use your platform's native atob() and btoa() methods instead
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
[3/5] Fetching packages...
error Could not write file "/opt/outline/yarn-error.log": "EACCES: permission denied, open '/opt/outline/yarn-error.log'"
error An unexpected error occurred: "EACCES: permission denied, unlink '/opt/outline/node_modules/.yarn-integrity'".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
`

Any chance you could help me out on where to get started with solving this?
Ubuntu 24.04 with Docker on a Hetzner VPS

Thanks
Michael

**Comment by tommoor** [comment]
> No lockfile found

The first few words are concerning, and would result in the rest of these errors most likely. You should use the official docker images which certainly include the lockfile :)

**Reply by electric-m** [reply]
Well, thank you for your reply. I thought, that I WAS using the official image:

```yaml
version: "3"
services:
  outline:
    image: outlinewiki/outline:latest
    restart: always
    env_file: docker.env
    command: sh -c "yarn --env=production-ssl-disabled && yarn start --env=production-ssl-disabled"
```

I was following the tutorial here and used the example for the docke compose file:
https://thomasgriffin.com/how-to-install-the-outline-knowledge-base-wiki-on-ubuntu/

**Reply by electric-m** [reply]
Well, I now disabled this part:

```yaml
command: sh -c "yarn --env=production-ssl-disabled && yarn start --env=production-ssl-disabled"
```

Now at least outline ist starting up.
Now I need to figure out, how to login via E-Mail.

SMTP is configured, but I can only see the starting screen without to option to login via E-mail.
I read in the documentation, that I have to enable it, but how to do that without logging in beforehand?
For testing purposes, I was hoping to circumvent using OIDC or others...
<img width="594" alt="SCR-20240507-jhwc" src="https://github.com/outline/outline/assets/40071240/f621c3e5-0678-47c5-8bae-ef683d62caf6">

**Reply by Its4Nik** [reply]
So my idea was to just use googles OpenID Auth to create my first user and then enable SMTP inside the settings. I have a guide on that on my github:

https://github.com/its4nik/easyoutline

There is also a small tutorial on how to setup everything

Edit:

It is using a custom script to setup everything but if you have a look at it you should figure it out, if you still have problems you could open a discussion on my repo, I am more or less always online and there to help :)
